src/quartz/physics/collider/Collider.hpp

```diff
@@ -21,21 +25,28 @@ class quartz::physics::Collider {
     static Collider createBoxCollider(
         quartz::managers::PhysicsManager& physicsManager,
         reactphysics3d::RigidBody* p_rigidBody,
-        const quartz::physics::BoxCollider::Parameters& parameters
+        const quartz::physics::BoxShape::Parameters& boxShapeParameters
     );
     static Collider createSphereCollider(
         quartz::managers::PhysicsManager& physicsManager,
         reactphysics3d::RigidBody* p_rigidBody,
-        const quartz::physics::SphereCollider::Parameters& parameters
+        const quartz::physics::SphereShape::Parameters& sphereShapeParameters
     );
     
 public: // member functions
     Collider(const Collider& other) = delete;
     Collider(Collider&& other);
 
-    reactphysics3d::CollisionShape* getCollisionShapePtr();
+    USE_LOGGER(COLLIDER);
+
+    const reactphysics3d::CollisionShape* getCollisionShapePtr() const;
+    const reactphysics3d::Collider* getColliderPtr() const { return mp_collider; }
```

Comment:
we should figure out what we want to do with this paradigm of exposing the raw `reactphysics3d` pointers. We should think about this within the context of the `RigidBody`, the `Field`, and the `BoxShape` and `SphereShape` as well